pkg/resources/secret_with_oauth_authorization_code_grant.go

```diff
@@ -0,0 +1,218 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	secret, err := client.Secrets.ShowByID(ctx, id)
+	if secret == nil || err != nil {
+		if errors.Is(err, sdk.ErrObjectNotFound) {
+			d.SetId("")
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Warning,
+					Summary:  "Failed to retrieve secret. Target object not found. Marking the resource as removed.",
+				},
+			}
+		}
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to retrieve secret.",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+			},
+		}
+	}
+	secretDescription, err := client.Secrets.Describe(ctx, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = setStateToValuesFromConfig(d, secretAuthorizationCodeGrantSchema, []string{"oauth_refresh_token_expiry_time"}); err != nil {
+		return diag.FromErr(err)
+	}
+	if err = d.Set("api_authentication", secretDescription.IntegrationName); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := handleSecretRead(d, id, secret, secretDescription); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func UpdateContextSecretWithAuthorizationCodeGrant(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if request := handleSecretUpdate(id, d); request != nil {
+		if err := client.Secrets.Alter(ctx, request); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if d.HasChange("oauth_refresh_token") {
+		refreshToken := d.Get("oauth_refresh_token").(string)
+
+		request := sdk.NewAlterSecretRequest(id)
+		setRequest := sdk.NewSetForOAuthAuthorizationFlowRequest().WithOauthRefreshToken(refreshToken)
+		request.WithSet(*sdk.NewSecretSetRequest().WithSetForOAuthAuthorizationFlow(*setRequest))
+
+		if err := client.Secrets.Alter(ctx, request); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if d.HasChange("oauth_refresh_token_expiry_time") {
+		refreshTokenExpiryTime := d.Get("oauth_refresh_token_expiry_time").(string)
+
+		request := sdk.NewAlterSecretRequest(id)
+		setRequest := sdk.NewSetForOAuthAuthorizationFlowRequest().WithOauthRefreshTokenExpiryTime(refreshTokenExpiryTime)
+		request.WithSet(*sdk.NewSecretSetRequest().WithSetForOAuthAuthorizationFlow(*setRequest))
+
+		if err := client.Secrets.Alter(ctx, request); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return ReadContextSecretWithAuthorizationCodeGrant(ctx, d, meta)
+}
+
+func DeleteContextSecretWithAuthorizationCodeGrant(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := client.Secrets.Drop(ctx, sdk.NewDropSecretRequest(id).WithIfExists(*sdk.Bool(true))); err != nil {
```

Comment:
It could be simply `WithIfExists(true)`. This applies to other resources.

Reply:
changed for every resource

Reply:
changed